File: 1509-minimum-difference-between-largest-and-smallest-value-in-three-moves/1509-minimum-difference-between-largest-and-smallest-value-in-three-moves.cpp
class Solution {
public:
    int minDifference(vector<int>& nums) {
        if(nums.size() <= 4) return 0;
        
        int N = nums.size();
        sort(nums.begin(), nums.end());
        long answer = nums[N - 1] - nums[0];
        
        for(int i = 0; i <= 3; i++) {
            if(answer > abs(nums[0 + i] - nums[N - 1 - (3 - i)])) {
                answer = abs(nums[0 + i] - nums[N - 1 - (3 - i)]);
            }
        }
        
        return answer;
    }
};

File: 1509-minimum-difference-between-largest-and-smallest-value-in-three-moves/1509-minimum-difference-between-largest-and-smallest-value-in-three-moves.py
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        if len(nums) <= 4:
            return 0
        
        nums.sort()
        return min(abs(nums[0 + i] - nums[-1 - (3 - i)]) for i in range(4))